cpp/src/arrow/buffer.cc

```diff
@@ -47,6 +49,17 @@ Status Buffer::Copy(const int64_t start, const int64_t nbytes,
   return Copy(start, nbytes, default_memory_pool(), out);
 }
 
+std::string Buffer::ToHexString() {
+  const uint8_t* _data = is_mutable() ? mutable_data() : data();
+  std::stringstream ss;
+
+  for (int64_t i = 0; i < size(); ++i) {
+    ss << std::setfill('0') << std::setw(2) <<
```

Comment:
Instead you should be able to use the `HexEncode` function from `arrow/util/string.h`.

Reply:
Thanks @pitrou. `ToHexString` is using `HexEncode` now.